### PR TITLE
Set - add throws for insert- and createFromJson- methods

### PR DESCRIPTION
**Pull request by edualonso**
Just like with dictionaries, we don't support those methods in the first release.

This required moving all the `RealmSet` fields present in `AllTypes` to a separate class, `SetAllTypes` and `SetAllTypesPrimaryKey`. This had ramifications across all set testers.

I've skipped adding set fields to the annotation processor `AllTypes` asset file since it has already been done for dictionaries and it worked so the hassle of dealing with it outweighs the benefits of doing it - see https://jira.mongodb.org/browse/RJAVA-921 for more details on this.

### CODE REVIEW

File: realm/realm-library/src/androidTest/kotlin/io/realm/SetMiscTests.kt [hunk 1]
```diff
@@ -242,4 +239,24 @@ class SetMiscTests {
...
+
+    @Test
+    fun insertOrUpdate_unsupportedOperation() {
+        realm = Realm.getInstance(configFactory.createConfiguration())
+        realm.executeTransaction {
+            assertFailsWith<UnsupportedOperationException> {
+                realm.insertOrUpdate(SetContainerMigrationClass())
+            }
+        }
+    }
```
Comment (rorbech) [inline realm/realm-library/src/androidTest/kotlin/io/realm/SetMiscTests.kt:261]: For completion maybe also add test for the json methods.

File: realm/realm-library/src/testUtils/java/io/realm/entities/AllTypes.java [hunk 2]
```diff
@@ -108,36 +108,36 @@
...
-    @Required private RealmSet<Long> columnRequiredLongSet;
-    @Required private RealmSet<Short> columnRequiredShortSet;
-    @Required private RealmSet<Double> columnRequiredDoubleSet;
-    @Required private RealmSet<Byte> columnRequiredByteSet;
-    @Required private RealmSet<byte[]> columnRequiredBinarySet;
-    @Required private RealmSet<Date> columnRequiredDateSet;
-    @Required private RealmSet<ObjectId> columnRequiredObjectIdSet;
-    @Required private RealmSet<UUID> columnRequiredUUIDSet;
-    @Required private RealmSet<Decimal128> columnRequiredDecimal128Set;
+//    private RealmSet<Boolean> columnBooleanSet;
```
Comment (rorbech) [inline realm/realm-library/src/testUtils/java/io/realm/entities/AllTypes.java:111]: Should be remove? Or maybe I don't get the PR comment, but if there is any reason for keeping these, please also add a comment here.

**Review by rorbech — APPROVED** [review]
LGTM. Only a few very minor comments.

File: realm/realm-library/src/testUtils/java/io/realm/entities/AllTypes.java (hunk 2 above)
Comment (edualonso) [reply realm/realm-library/src/testUtils/java/io/realm/entities/AllTypes.java:111]: These are leftovers, forgot to remove them.

**Comment by edualonso** [comment]
Updated the PR with comments. Tests are passing, merging at once 🚀

**Pull request merged**